### PR TITLE
Bump `aiohttp` to `3.11.1`

### PR DESCRIPTION
**Pull request by kaxil**
https://github.com/aio-libs/aiohttp/issues/9866 was fixed by https://github.com/pnuckowski/aioresponses/pull/262#issuecomment-2476728212

https://pypi.org/project/aiohttp/3.11.1/

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by vincbeck — APPROVED** [review]

**Comment by kaxil** [comment]
https://github.com/pnuckowski/aioresponses/pull/262 hasn't been merged yet though

**Comment by potiuk** [comment]
> [pnuckowski/aioresponses#262](https://github.com/pnuckowski/aioresponses/pull/262) hasn't been merged yet though

It's unllikely to be merged - the maintainer seems to be MIA for 2 years. That's why likely aiohttp brought back compatiblity with released version of the aioresponses as it is popular

File: providers/src/airflow/providers/http/provider.yaml [hunk 1]
```diff
@@ -68,8 +68,7 @@ dependencies:
   # release it as a requirement for airflow
   - requests>=2.27.0,<3
   - requests-toolbelt>=0.4.0
-  # Limit temporarily aiohttp to <3.11.0 because of https://github.com/aio-libs/aiohttp/issues/9866
-  - aiohttp>=3.9.2,<3.11.0
+  - aiohttp>=3.11.1
```
Comment (potiuk) [inline providers/src/airflow/providers/http/provider.yaml]:
```suggestion
  - aiohttp>=3.9.2,!=3.11.0
```

Comment (potiuk) [reply providers/src/airflow/providers/http/provider.yaml]: I think that would be better. There might be abug found in 3.11.1 - seems a bit dangerous to bump lower bounds that fast.

File: generated/provider_dependencies.json [hunk 2]
```diff
@@ -751,7 +751,7 @@
   },
   "http": {
     "deps": [
-      "aiohttp>=3.9.2,<3.11.0",
+      "aiohttp>=3.11.1",
```
Comment (kaxil) [inline generated/provider_dependencies.json]:
```suggestion
      "aiohttp>=3.9.2,!=3.11.0",
```

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
looks good. Maybe one nit - would be great to add comment with link to the issue in provider.yaml

**Comment by potiuk** [comment]
(we have it in commit so no strictly necessary - but it safes git blame etc.)

File: providers/src/airflow/providers/http/provider.yaml [hunk 3]
```diff
@@ -68,8 +68,7 @@ dependencies:
   # release it as a requirement for airflow
   - requests>=2.27.0,<3
   - requests-toolbelt>=0.4.0
-  # Limit temporarily aiohttp to <3.11.0 because of https://github.com/aio-libs/aiohttp/issues/9866
-  - aiohttp>=3.9.2,<3.11.0
+  - aiohttp>=3.9.2,!=3.11.0
```
Comment (kaxil) [inline providers/src/airflow/providers/http/provider.yaml:72]:
```suggestion
  # aiohttp 3.11.0 had incompatible change: https://github.com/aio-libs/aiohttp/issues/9866
  - aiohttp>=3.9.2,!=3.11.0
```

**Comment by kaxil** [comment]
> (we have it in commit so no strictly necessary - but it safes git blame etc.)

Done

**Review by potiuk — APPROVED** [review]

**Review by jscheffl — APPROVED** [review]

**Comment by kaxil** [comment]
Damn! last commit msg was used as commit title :(

**Comment by potiuk** [comment]
Good we have the comment :)

**Pull request merged**